### DISCUSSION TITLE
Flashing the AI Decks boot loader.

### DISCUSSION THREAD

**Discussion by omsingh-rgb** [bitcraze/discussions#1221] [Q&A]
Hello all,

I have recently got a crazyfile and a AI deck for a project that is coming to completion soon. I have tried flashing the WiFi example from https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/ however it is getting to 99% and stops. I am now assuming that the boot loader needs to be updated. 

I have on me a ST Link V2 and unfortunately don't have the recommended programmers. Is there any way I can still update the boot loader as it is not in the budget/time constraints to get another programmer. I am also tempted to create one with a Pico or something but I am completely unsure how I would have it compatible with OpenOCD. 

Any advice/direction will be appreciated.

**Comment by gemenerik** [comment]
I'm afraid [the ST Link will not work](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/infrastructure/jtag-programmer/).  @Yaooooo from GreenWaves [mentioned a hardware hack](https://forum.bitcraze.io/viewtopic.php?t=4485) as a possible workaround for debugging the RISC-V cores, you could try with an [issue here](https://github.com/GreenWaves-Technologies/gap_sdk).

But realistically, you have time constraints, so I would try to free some budget for an Olimex.
